draft-irtf-cfrg-hpke.md

```diff
@@ -271,52 +273,108 @@ A set of algorithm identifiers for concrete instantiations of these
 primitives is provided in {{ciphersuites}}.  Algorithm identifier
 values are two bytes long.
 
+The following two functions are defined for a KDF to facilitate domain
+separation of calls as well as context binding:
+
+~~~
+def LabeledExtract(salt, label, IKM):
+  labeled_IKM = concat("RFCXXXX ", label, IKM)
+  return Extract(salt, labeled_IKM)
```

Comment:
Is there a reason to put the label on the IKM vs. the salt?

Reply:
In the end, the goal of the label is to separate the input domains of all HMAC calls. This concerns both Extract and Expand. The three Expand calls for key, nonce, exporter_secret already have a label as prefix of the second argument, and that's why it's in the second argument for Extract, too (and thus attached to the IKM).